sdk/core/azure-core/azure/core/settings.py

```diff
@@ -229,19 +233,27 @@ class PrioritizedSetting:
     :param callable convert: a function to convert values before they are returned
     """
 
-    def __init__(self, name: str, env_var: Optional[str] = None, system_hook=None, default=_Unset, convert=None):
+    def __init__(
+        self,
+        name: str,
+        env_var: Optional[str] = None,
+        system_hook: Optional[Callable[[], ValidInputType]] = None,
+        default: Union[ValidInputType, _Unset] = _unset,
```

Comment:
We document a valid default as being str, int or float - but we're not typing that here?

Reply:
Though it seems to be incomplete - as within this file it seems that `default` also supports `None` and `bool`

Reply:
I typed `default` a generic, so implementer of the `PrioritizedSetting` protocol can specify the precise type they support as in put. Technically, nothing in the `PrioritizedSetting` code forces a specific type as we sub-contract the processing of the input to a callback.

Reply:
Wondering if we should update that docstring then?

Reply:
This one also becomes quite interesting as default is `Union[ValidInputType, _Unset]` but is passed in to `self._convert` which is `Callable[[Union[ValidInputType, str]], ValueType]` , which wont match. Same for `self._user`. Does convert need to be made generic as well ?

Reply:
There is no mismatch, as the `_convert` is called only if this is not unset, which mean the revelaed type as the time of calling the callback is `ValidInputType`, which is compatible with `Union[ValueInputType, str]`

Reply:
I updated the docstring